simplesat/dependency_solver.py

```diff
@@ -65,12 +71,62 @@ def _create_rules(self, request):
 
         rules_generator = RulesGenerator(pool, request, installed_map)
 
-        return list(rules_generator.iter_rules())
+        return requirement_ids, list(rules_generator.iter_rules())
+
+
+def _connected_packages(solution, requirement_ids, pool):
+    """ Return packages which are associated with a requirement. """
+
+    # Our strategy is as follows:
+    # ... -> pkg.dependencies -> pkg strings -> ids -> _id_to_package -> ...
+
+    # We only need to identify packages which will be installed
+    sol_set = set(s for s in solution if s > 0)
+
+    # This dict can recover ids from the strings produced by pkg.dependencies
+    package_string_to_id = {}
+    for pkg_id in sol_set:
+        pkg = pool._id_to_package[pkg_id]
+        pkg_key = pkg.name
+        package_string_to_id[pkg_key] = pkg_id
+
+    def neighborfunc(pkg_id):
+        """ Given a pkg id, return the pkg ids of the dependencies that
+        appeared in our solution. """
+        dep_strings = pool._id_to_package[pkg_id].dependencies
+        pkg_keys = (d.split()[0] for d in dep_strings)
```

Comment:
@johntyree the dichotomy is gone in the newer enstaller. It is called legacy because the `Requirement` class can handle much richer semantics, and the goal is to support those richer semantics in next egg format (but we need the new solver first :) )